src/main/java/io/cdap/plugin/function/DiscretePercentile.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.function;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.expressions.MutableAggregationBuffer;
+import org.apache.spark.sql.expressions.UserDefinedAggregateFunction;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.MapType;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+/**
+ *  Calculate Discrete Percentile
+ */
+public class DiscretePercentile extends UserDefinedAggregateFunction {
+
+  private final double percentile;
+  private final DataType inputSchemaSparkType;
+
+  public DiscretePercentile(double percentile, Schema.Type inputSchemaType) {
+    this.percentile = percentile;
+    this.inputSchemaSparkType = convertType(inputSchemaType);
+  }
+
+  @Override
+  public StructType inputSchema() {
+    List<StructField> inputFields = new ArrayList<>();
+    inputFields.add(DataTypes.createStructField("value", inputSchemaSparkType, true));
+    return DataTypes.createStructType(inputFields);
+  }
+
+  private DataType convertType(Schema.Type type) {
+    switch (type) {
+      case INT:
```

Comment:
INT and LONG could also represent logical types date, timestamp. We should make sure we are not applying thiis on those types